test/test_auth.py

```diff
@@ -0,0 +1,251 @@
+        def _replace_attr(name, value):
+            new_cfg = orig_cfg.replace(**{name: value})
+            self.assertIsNot(orig_cfg, new_cfg)  # must return new object
+
+            self.assertEqual(value, getattr(new_cfg, name))  # must replace specified value
+
+            # check that only the one attribute differs
+            for attr in awscrt.auth.AwsSigningConfig._attributes:
+                if attr == name:
+                    self.assertNotEqual(getattr(orig_cfg, attr), getattr(new_cfg, attr),
+                                        "replaced value should not match original")
+                else:
+                    self.assertEqual(getattr(orig_cfg, attr), getattr(new_cfg, attr),
+                                     "value should match original")
+
+        _replace_attr('algorithm', awscrt.auth.AwsSigningAlgorithm.Sigv4Header)
+        _replace_attr('credentials_provider',
+                      awscrt.auth.StaticAwsCredentialsProvider(EXAMPLE_ACCESS_KEY_ID, EXAMPLE_SECRET_ACCESS_KEY))
+        _replace_attr('region', 'us-west-2')
+        _replace_attr('service', 'aws-nothing-but-bees')
+        _replace_attr('date', datetime.datetime(year=2001, month=1, day=1))
+        _replace_attr('should_sign_param', lambda x: True)
+        _replace_attr('use_double_uri_encode', False)
+        _replace_attr('should_normalize_uri_path', True)
+        _replace_attr('sign_body', True)
+
+        # check that we can replace multiple values at once
+        new_cfg = orig_cfg.replace(region='us-west-3', service='aws-slow-blinking')
+        self.assertEqual('us-west-3', new_cfg.region)
+        self.assertEqual('aws-slow-blinking', new_cfg.service)
+        self.assertEqual(orig_cfg.should_sign_param, new_cfg.should_sign_param)
+
+
+# Test values copied from aws-c-auth/tests/aws-sig-v4-test-suite/get-vanilla"
+SIGV4TEST_ACCESS_KEY_ID = 'AKIDEXAMPLE'
+SIGV4TEST_SECRET_ACCESS_KEY = 'wJalrXUtnFEMI/K7MDENG+bPxRfiCYEXAMPLEKEY'
+SIGV4TEST_SESSION_TOKEN = '6e86291e8372ff2a2260956d9b8aae1d763fbf315fa00fa31553b73ebf194267'
+SIGV4TEST_SERVICE = 'service'
+SIGV4TEST_REGION = 'us-east-1'
+SIGV4TEST_METHOD = 'GET'
+SIGV4TEST_PATH = '/'
+SIGV4TEST_UNSIGNED_HEADERS = [
+    ('Host', 'example.amazonaws.com'),
+    ('X-Amz-Date', '20150830T123600Z')]
+SIGV4TEST_SIGNED_HEADERS = [
+    ('Host',
+     'example.amazonaws.com'),
+    ('X-Amz-Date',
+     '20150830T123600Z'),
+    ('Authorization',
+     'AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, Signature=5fa00fa31553b73ebf1942676e86291e8372ff2a2260956d9b8aae1d763fbf31')]
+
+
+class TestSigner(NativeResourceTest):
+
+    def test_create(self):
+        signer = awscrt.auth.AwsSigner()
+
+    def test_signing_sigv4_headers(self):
+        signer = awscrt.auth.AwsSigner()
+
+        credentials_provider = awscrt.auth.StaticAwsCredentialsProvider(
+            SIGV4TEST_ACCESS_KEY_ID, SIGV4TEST_SECRET_ACCESS_KEY, SIGV4TEST_SESSION_TOKEN)
```

Comment:
Something's wrong here, because your session token isn't making it into the final header results

Reply:
WHOOPS. yes. shouldn't be using session key. This whole test was broken because I was comparing against the wrong thing. Fixed now